lib/kb_Metrics/metricsDBs.py

```diff
@@ -106,17 +109,20 @@ def insert_activity_records(self, mt_docs):
         Insert an iterable of user activity documents
         """
         if not isinstance(mt_docs, list):
-            raise ValueError('The variable mt_docs must be a list of mutable mapping type data.')
+            raise ValueError('Variable mt_docs must be' +
+                             ' a list of mutable mapping type data.')
 
         # grab handle(s) to the database collection(s) targeted
-        self.mt_act = self.metricsDBs['metrics'][MongoMetricsDBI._MT_DAILY_ACTIVITIES]
+        self.mt_act = self.metricsDBs['metrics'][
+                            MongoMetricsDBI._MT_DAILY_ACTIVITIES]
         insert_ret = None
         try:
             # get an instance of InsertManyResult(inserted_ids, acknowledged)
-            insert_ret = self.mt_act.insert_many(mt_docs, ordered=True)
+            insert_ret = self.mt_act.insert_many(mt_docs, ordered=False)
         except BulkWriteError as bwe:
-            # skip uplicate key error (code=11000)
-            panic = filter(lambda x: x['code'] != 11000, bwe.details['writeErrors'])
+            # skip duplicate key error (code=11000)
+            panic = filter(lambda x: x['code'] != 11000,
+                           bwe.details['writeErrors'])
             if len(panic) > 0:
```

Comment:
If it throws BulkWriteError exceptions, it will be caught if the error code is not '11000'. Empty panic means it all exceptions are 11000 (duplicated records).

Reply:
what I mean is if BulkWriteError and error code is 11000, program will not throw anything. Is that expected?

Reply:
yes.

Reply:
modified and tested.